### PR TITLE
Add Implementation to Tokenizer Module and OwnableERC20

### PR DESCRIPTION
**Pull request by sebsadface**
## Description

This PR introduces implementations and interfaces for the new `TokenizerModule` and `OwnableERC20` contracts, as outlined in #143.

### New Contracts

  - `IOwnableERC20`: Defines the standard interface for ERC20 templates used in the `TokenizerModule`.  
  - `ITokenizerModule`: Specifies the interface for the `TokenizerModule`.  

  - `OwnableERC20`: A base implementation of `IOwnableERC20`. This contract is integrated with the `TokenizerModule` via the Beacon Proxy pattern. When the `TokenizerModule` fractionalizes an IP, it deploys a Beacon Proxy instance of the `OwnableERC20` template, and ties each instance to a specific IP.  
  - `TokenizerModule`: Manages the fractionalization of IP assets (IPA). It allows IP owners to select from whitelisted `OwnableERC20` templates for their fractionalized tokens and enables developers to whitelist new templates for others to use.

### Tests

Test cases have been added for all functions in both contracts. All tests pass locally.

### CODE REVIEW

File: contracts/interfaces/modules/tokenizer/IOwnableERC20.sol [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    /// @param to The address to mint tokens to
+    /// @param amount The amount of tokens to mint
+    function mint(address to, uint256 amount) external;
+
+    /// @notice Returns the upgradable beacon
+    function upgradableBeacon() external view returns (address);
+
+    /// @notice Returns whether the contract supports an interface
+    /// @param interfaceId The interface ID
+    function supportsInterface(bytes4 interfaceId) external view returns (bool);
```
Comment (kingster-will) [inline contracts/interfaces/modules/tokenizer/IOwnableERC20.sol:35]: We might not need define this interface, just inherit from IERC165.

File: contracts/modules/tokenizer/TokenizerModule.sol [hunk 2]
```diff
@@ -1,38 +1,135 @@
...
+    function _isExpiredNow(address ipId) internal view returns (bool) {
+        uint256 expireTime = _getExpireTime(ipId);
+        return expireTime != 0 && expireTime < block.timestamp;
+    }
 
+    /// @dev Get the expiration time of an IP
+    /// @param ipId The address of the IP
+    function _getExpireTime(address ipId) internal view returns (uint256) {
+        return IIPAccount(payable(ipId)).getUint256(EXPIRATION_TIME);
+    }
```
Comment (kingster-will) [inline contracts/modules/tokenizer/TokenizerModule.sol:118]: The function is duplicate with LicenseRegistry. just reuse the function `isExpiredNow` from the LicenseRegistry would be better.

File: contracts/modules/tokenizer/TokenizerModule.sol [hunk 3]
```diff
@@ -1,38 +1,135 @@
...
+    /// @notice Tokenizes an IP
+    /// @param ipId The address of the IP
+    /// @param tokenTemplate The address of the token template
+    /// @param initData The initialization data for the token
+    /// @return token The address of the newly created token
+    function tokenize(
+        address ipId,
+        address tokenTemplate,
+        bytes calldata initData
+    ) external verifyPermission(ipId) returns (address token) {
```
Comment (kingster-will) [inline contracts/modules/tokenizer/TokenizerModule.sol:88]: add one more check, that revert if the ipId has already been tokenized.

File: contracts/interfaces/modules/tokenizer/IOwnableERC20.sol (hunk 1 above)
Comment (sebsadface) [reply contracts/interfaces/modules/tokenizer/IOwnableERC20.sol]: resolved in [8efd8a2](https://github.com/storyprotocol/protocol-periphery-v1/pull/144/commits/8efd8a2d1a7e91d1a982da3735b9fa1cde96d4b4)

File: contracts/modules/tokenizer/TokenizerModule.sol (hunk 3 above)
Comment (sebsadface) [reply contracts/modules/tokenizer/TokenizerModule.sol]: resolved in [8efd8a2](https://github.com/storyprotocol/protocol-periphery-v1/pull/144/commits/8efd8a2d1a7e91d1a982da3735b9fa1cde96d4b4)

File: contracts/modules/tokenizer/TokenizerModule.sol (hunk 2 above)
Comment (sebsadface) [reply contracts/modules/tokenizer/TokenizerModule.sol]: The problem is the `_isExpiredNow` is an internal function inside `LicenseRegistry`. Do we want to make it public for external use?

Comment (kingster-will) [reply contracts/modules/tokenizer/TokenizerModule.sol]: There is an external `isExpiredNow` in LicenseRegistry.
https://github.com/storyprotocol/protocol-core-v1/blob/34c5f2b10d3530129c91fb02bdfc4e4befaf354e/contracts/interfaces/registries/ILicenseRegistry.sol#L209

Comment (sebsadface) [reply contracts/modules/tokenizer/TokenizerModule.sol]: I see, thank you for pointing that out. Resolved in [e0d4867](https://github.com/storyprotocol/protocol-periphery-v1/pull/144/commits/e0d486761c0a70d3cd6dc4ef4c39d44cd3b39d92)

File: contracts/interfaces/modules/tokenizer/IOwnableERC20.sol [hunk 4]
```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: MIT
```
Comment (kingster-will) [inline contracts/interfaces/modules/tokenizer/IOwnableERC20.sol:1]: Use BUSL license

**Review by kingster-will — APPROVED** [review]
LGTM. Please use BUSL license for the files of Module related code.

File: contracts/interfaces/modules/tokenizer/ITokenizerModule.sol [hunk 5]
```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: MIT
```
Comment (letsvitta) [inline contracts/interfaces/modules/tokenizer/ITokenizerModule.sol:1]: MIT -> BUSL-1.1

File: contracts/modules/tokenizer/OwnableERC20.sol [hunk 6]
```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MIT
```
Comment (letsvitta) [inline contracts/modules/tokenizer/OwnableERC20.sol:1]: BUSL-1.1

File: contracts/modules/tokenizer/TokenizerModule.sol [hunk 7]
```diff
@@ -1,38 +1,160 @@
-// SPDX-License-Identifier: BUSL-1.1
+// SPDX-License-Identifier: MIT
```
Comment (letsvitta) [inline contracts/modules/tokenizer/TokenizerModule.sol:1]: BUSL-1.1

File: script/utils/DeployHelper.sol [hunk 8]
```diff
@@ -894,6 +942,8 @@ contract DeployHelper is
 
         // add evenSplitGroupPool to whitelist of group pools
         groupingModule.whitelistGroupRewardPool(address(evenSplitGroupPool), true);
+
+        disputeModule.whitelistDisputeTag("PLAGIARISM", true);
```
Comment (letsvitta) [inline script/utils/DeployHelper.sol:946]: Why do we need to whitelist dispute tag here?

Comment (kingster-will) [inline script/utils/DeployHelper.sol:946]: Why do we need to whitelist dispute tag here?

File: contracts/interfaces/modules/tokenizer/IOwnableERC20.sol (hunk 4 above)
Comment (sebsadface) [reply contracts/interfaces/modules/tokenizer/IOwnableERC20.sol]: Resolved in [f90eb33](https://github.com/storyprotocol/protocol-periphery-v1/pull/144/commits/f90eb3335d7e1c5e9138b4fd4e0b7ffe1786fc18)

File: script/utils/DeployHelper.sol (hunk 8 above)
Comment (sebsadface) [reply script/utils/DeployHelper.sol]: It's for the testing that tokenizer module will revert if the IP is tagged. Moved this line to the test function instead in [f90eb33](https://github.com/storyprotocol/protocol-periphery-v1/pull/144/commits/f90eb3335d7e1c5e9138b4fd4e0b7ffe1786fc18) for better readability :)

**Pull request merged**